### PR TITLE
Topics index

### PR DESCRIPTION
**Pull request by RCarmona53**
#### Board:
* [Ticket #RSLC-1]([RSLC-1](https://trello.com/c/4NnOTkyu/16-rslc-1-as-a-user-i-should-be-able-to-see-all-the-topics-in-the-system))
---
#### Description:
As a user I should be able to see all the topics in the system
---
#### Tasks:
A topic should contain:
- name
- image

Endpoint example: [Target MVD (?) · Apiary](https://targetapi2021.docs.apiary.io/#reference/0/topics-collection/get-topics)

### CODE REVIEW

File: app/controllers/api/v1/topics_controller.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module Api
+  module V1
+    class TopicsController < ApplicationController
+      def index
+        @topics = policy_scope(Topic).includes(image_attachment: :blob)
+        respond_to do |format|
+          format.json
+        end
```
Comment (enzofab91) [inline app/controllers/api/v1/topics_controller.rb:8]: If you are only expecting to return json views you can do something like

```suggestion
        render json: @topics
```

File: db/migrate/20230519173927_create_topics.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class CreateTopics < ActiveRecord::Migration[7.0]
+  def change
+    create_table :topics do |t|
+      t.string :name, null: false, unique: true
+
+      t.timestamps
```
Comment (enzofab91) [inline db/migrate/20230519173927_create_topics.rb]:
```suggestion
      t.timestamps, null: false
```

File: db/migrate/20230522135216_add_index_to_topics.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddIndexToTopics < ActiveRecord::Migration[7.0]
+  def change
+    add_index :topics, :name, unique: true
```
Comment (enzofab91) [inline db/migrate/20230522135216_add_index_to_topics.rb]: This can be added in the other migration file and have only one, wdyt?

**Review by enzofab91 — APPROVED** [review]
Nice work! 💯

File: db/migrate/20230522135216_add_index_to_topics.rb (hunk 3 above)
Comment (RCarmona53) [reply db/migrate/20230522135216_add_index_to_topics.rb]: Done!

File: db/migrate/20230519173927_create_topics.rb (hunk 2 above)
Comment (RCarmona53) [reply db/migrate/20230519173927_create_topics.rb]: Done!

**Review by santiagodiaz — APPROVED** [review]
Nice work! 👏

File: app/controllers/api/v1/topics_controller.rb (hunk 1 above)
Comment (RCarmona53) [reply app/controllers/api/v1/topics_controller.rb:8]: Enzo this is how postman do the GET /api/v1/topics
"topics": [
        {
            "id": 1,
            "name": "Futbol",
            "image": {
                "name": "image",
                "record": {
                    "id": 1,
                    "name": "Futbol",
                    "created_at": "2023-05-24T13:17:24.665Z",
                    "updated_at": "2023-05-24T13:17:24.691Z"
                }
            },
            "created_at": "2023-05-24T13:17:24.665Z",
            "updated_at": "2023-05-24T13:17:24.691Z"
        }

And this is where i call the @topics controller
json.topics do |json|
  json.array! @topics, :id, :name, :image, :created_at, :updated_at
end
So i resolve the comment without using the render.

**Pull request merged**